src/logger/logger.ts

```diff
@@ -1,12 +1,100 @@
-import pino from "pino"
+import _ from "lodash"
+import pino, { Logger as PinoLogger } from "pino"
+
+import { StackFrame, get, parse } from "./stack-trace"
+
+interface WarnMeta {
+  params: Record<string, unknown>
+}
+
+interface ErrorMeta {
+  error: unknown
+}
+
+const MAX_STACK_DEPTH = 5
+
+export interface TraceMeta {
+  file: string
+  line: number
+  func: string
+}
+
+export const getTraceMeta = (frames: StackFrame[]): TraceMeta[] =>
+  frames.map((frame) => ({
+    file: frame.getFileName(),
+    line: frame.getLineNumber(),
+    func: frame.getFunctionName(),
+  }))
+
+export class Logger {
+  _logger: PinoLogger
+
+  constructor(logger: PinoLogger) {
+    this._logger = logger
+  }
+
+  public child = (
+    bindings: Record<string, unknown> & { module: string }
+  ): Logger => {
+    const child = this._logger.child(bindings)
+    return new Logger(child)
+  }
+
+  public error = (
+    message: string,
+    meta?: WarnMeta & ErrorMeta & Record<string, unknown>
+  ): void => {
+    const stackFrames = meta?.error instanceof Error ? parse(meta.error) : get()
+    const stackTrace = getTraceMeta(stackFrames.slice(0, MAX_STACK_DEPTH))
+
+    this._logger.error(message, {
+      isomer: {
+        meta,
+        stackTrace,
+      },
+    })
+  }
+
+  public info = (message: string, meta?: Record<string, unknown>): void => {
+    if (_.isEmpty(meta)) {
+      this._logger.info(message)
+    } else {
+      this._logger.info(message, {
+        isomer: {
+          meta,
+        },
+      })
```

Comment:
⚠️ Does this call signature actually work? [Documentation](https://github.com/pinojs/pino/blob/master/docs/api.md#info) says it should be object first then message.

Quick test to confirm:

```javascript
pino = require('pino')
logger =   pino({
    errorKey: "error",
    messageKey: "message",
    formatters: {
      level: (label) => ({ level: label.toUpperCase() }),
    },
  })

logger.info('hello', {meta: {foo: 1}})
```
gives
```
{"level":"INFO","time":1711066498870,"pid":69756,"hostname":"Timothee-Groleaus-MacBook-Pro-14-inch-2021-","message":"hello"}
```

On the other hand, this works as expected:
```javascript
logger.info({meta: {foo: 1}}, 'hello')
```
gives
```
{"level":"INFO","time":1711066992960,"pid":72744,"hostname":"Timothee-Groleaus-MacBook-Pro-14-inch-2021-","meta":{"foo":1},"message":"hello"}
```

Reply:
gud ketch, fixed!